pyci/fanci/fanci.py

```diff
@@ -227,6 +227,7 @@ def optimize(
         mode: str = "lstsq",
         use_jac: bool = False,
         sigma: float = 0.1,
+	    custom_optimizer: callable=None,
```

Comment:
Can you just use the mode keyword instead of `custom(_optimizer)`? `mode` can be `("lstsq" | "root" | Callable)`, making `custom(_optimizer)` unneccessary.

Then,
```python3

if mode == "lstsq":
    pass
elif mode == "root":
    pass
elif callable(mode):
    pass
```